src/main/java/Applicaiton.java

```diff
@@ -0,0 +1,214 @@
+                saveCurrentCommand();
+                currentCommand = getCommandFromString(inputCommand);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * This method saves the current state (command)
+     */
+    private void saveCurrentCommand() {
+        previousCommand = currentCommand;
+    }
+
+    /**
+     * This method changes current state (command) to the previous state (command)
+     */
+    private void changeCommandToPrev() {
+        currentCommand = previousCommand;
+    }
+
+    /**
+     * This method is the running method of the application. It's purpose is to switch between commands and execute
+     * audio player methods
+     *
+     * @throws IOException
+     */
+    public void run() throws IOException, InterruptedException {
+        while (currentCommand != EXIT) {
+            switch (currentCommand) {
+                case PLAY:
+                    audioPlayer.playSong();
+                    break;
+                case NEXT:
+                    audioPlayer.nextSong();
+                    audioPlayer.playSong();
+                    break;
+                case PREVIOUS:
+                    audioPlayer.prevSong();
+                    audioPlayer.playSong();
+                    break;
+                case REPLAY:
+                    audioPlayer.replay();
+                    audioPlayer.playSong();
+                    break;
+                case SHUFFLE:
+                    audioPlayer.shuffle();
+                    break;
+                case PAUSE:
+                    System.out.println("On pause..");
+                    audioPlayer.pauseSong();
+                    break;
+                case SIZE:
+                    showSize();
+                    break;
+                case ADD:
+                    showAddDialog();
+                    break;
+                case SEARCH_BY_TITLE:
+                    showSearchByTitleDialog();
+                    break;
+                case SEARCH_BY_SINGER:
+                    showSearchBySingerDialog();
+                    break;
+            }
+        }
+    }
+
+    /**
+     * This method visualize dialog to allow the user to search for songs that match a singer.
+     * The method waits for user input, if the input is valid the searching is starting. The method returns appropriate message
+     * to the user if the there aren't any records or the input of the user is invalid. The method also return the application to
+     * * previous state, which was before the searching.
+     */
+    private void showSearchBySingerDialog() {
+        String result, userInput;
+        Scanner consoleInput = new Scanner(System.in);
+
+        System.out.print("Enter singer: ");
+        userInput = consoleInput.nextLine();
+        if (userInput != null && !userInput.isEmpty()) {
+            result = audioPlayer.searchBySinger(new Singer(userInput));
```

Comment:
Instead of creating `new Singer` object - change the method `searchBySinger(Singer)` to `searchBySingerName(String)`